### PR TITLE
Move GKE tests to cloud/gcp package

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/kubernetes/issues/98326

```release-note
NONE
```

/kind cleanup
/priority important-soon

### CODE REVIEW

File: test/e2e/e2e_test.go [hunk 1]
```diff
@@ -45,6 +45,7 @@ import (
 	_ "k8s.io/kubernetes/test/e2e/auth"
 	_ "k8s.io/kubernetes/test/e2e/autoscaling"
 	_ "k8s.io/kubernetes/test/e2e/cloud"
+	_ "k8s.io/kubernetes/test/e2e/cloud/gcp"
```
Comment (spiffxp) [inline test/e2e/e2e_test.go:48]: This is already handled by `test/e2e/cloud/imports.go`, don't need to add here

Comment (wojtek-t) [reply test/e2e/e2e_test.go:48]: done

**Review by navidshaikh — APPROVED** [review]
lgtm

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm

**Pull request merged**